src/main/java/org/opentripplanner/graph_builder/linking/StreetSplitter.java

```diff
@@ -1,11 +1,834 @@
+     *
+     * @param location
+     * @param options
+     * @param endVertex true if this is destination vertex
+     * @return
+     */
+    public Vertex linkOriginDestination(GenericLocation location, RoutingRequest options, boolean endVertex) {
+
+        if (endVertex) {
+            LOG.debug("Finding end vertex for {}", location);
+        } else {
+            LOG.debug("Finding start vertex for {}", location);
+        }
+        Coordinate coord = location.getCoordinate();
+        //TODO: add nice name
+        String name;
+
+        if (location.name == null || location.name.isEmpty()) {
+            if (endVertex) {
+                name = "Destination";
+            } else {
+                name = "Origin";
+            }
+        } else {
+            name = location.name;
+        }
+        TemporaryStreetLocation closest = new TemporaryStreetLocation(UUID.randomUUID().toString(),
+            coord, new NonLocalizedString(name), endVertex);
+
+        TraverseMode nonTransitMode = TraverseMode.WALK;
+        boolean linkWithSingleMode = true;
+
+        //It can be null in tests
+        if (options != null) {
+            TraverseModeSet modes = options.modes;
+            if (modes.getCar())
+                // for park and ride we will start in car mode and walk to the end vertex
+                if (endVertex && (options.parkAndRide || options.kissAndRide)) {
+                    nonTransitMode = TraverseMode.WALK;
+                }
+                // tnc routing and car rental it would be possible to walk or drive
+                else if (options.useTransportationNetworkCompany || options.allowCarRental) {
+                    linkWithSingleMode = false;
+                    if(!linkToGraph(
+                        closest,
+                        new TraverseModeSet(TraverseMode.WALK, TraverseMode.CAR),
+                        options,
+                        NON_DESTRUCTIVE_SPLIT,
+                        false
+                    )) {
+                        LOG.warn("Couldn't link {}", location);
+                    }
+                } else {
+                    nonTransitMode = TraverseMode.CAR;
+                }
+            else if (modes.getWalk())
+                nonTransitMode = TraverseMode.WALK;
+            else if (modes.getBicycle())
+                nonTransitMode = TraverseMode.BICYCLE;
+        }
+
+        if(linkWithSingleMode && !linkToGraph(closest, nonTransitMode, options, NON_DESTRUCTIVE_SPLIT, false)) {
+            LOG.warn("Couldn't link {}", location);
+        }
+        return closest;
+
+    }
+
+    public Boolean getAddExtraEdgesToAreas() {
+        return addExtraEdgesToAreas;
+    }
+
+    public void setAddExtraEdgesToAreas(Boolean addExtraEdgesToAreas) {
+        this.addExtraEdgesToAreas = addExtraEdgesToAreas;
+    }
+
+    public void removeSemiPermanentVerticesAndEdges (SemiPermanentSplitterVertex vertex) {
```

Comment:
This method needs javadoc. Something about how this will remove only the semipermanent edges connected to the splitter vertex.

Reply:
Will add.